File: cpp/problem_3.cpp
#include <iostream>
using namespace std;

int main()
{
  long int n = 600851475143;
  long int b = 27;
  int answer;
  while (b < 100000) {
    if (n % b == 0) { // if it is a factor
      for (int i = 1; i < b/2; i++) {
        if (i == (b/2 - 1)) {
          answer = b;
          cout << b << " is prime" << endl;
        }
        else if (b % i == 0) {
          cout << b << " is not prime" << endl;
        }
      }
    }
    b += 2;
  }
  cout << answer << endl;
  return 0;
}

File: ruby/problem3.rb
# The prime factors of 13195 are 5, 7, 13 and 29.
# What is the largest prime factor of the number 600851475143?
require 'prime'
n = 600851475143
start = Time.now
answer = (n**0.5).to_i.downto(3).select(&:odd?).detect do |e|
 # Prime.instance.prime? e
  (3..e**0.5).all? {|test_val| e%test_val!=0} and n%e == 0
end
puts answer
puts "took #{Time.now - start} seconds"
